src/coreclr/vm/frozenobjectheap.cpp

```diff
@@ -136,18 +136,27 @@ Object* FrozenObjectSegment::TryAllocateObject(PTR_MethodTable type, size_t obje
 {
     _ASSERT(m_pStart != nullptr && FOH_SEGMENT_SIZE > 0 && m_SegmentHandle != nullptr); // Expected to be inited
     _ASSERT(IS_ALIGNED(m_pCurrent, DATA_ALIGNMENT));
+    _ASSERT(objectSize <= FOH_COMMIT_SIZE);
```

Comment:
Where is this guaranteed? `AllocateString` calls this without any upfront check on `objectSize`.

Reply:
Ah ok, it is guaranteed in FrozenObjectHeapManager